private/smithy-rpcv2-cbor/src/protocols/Rpcv2cbor.ts

```diff
@@ -0,0 +1,1227 @@
+};
+
+/**
+ * deserializeRpcv2cborOptionalInputOutputCommand
+ */
+export const de_OptionalInputOutputCommand = async (
+  output: __HttpResponse,
+  context: __SerdeContext
+): Promise<OptionalInputOutputCommandOutput> => {
+  cr(output);
+  if (output.statusCode >= 300) {
+    return de_CommandError(output, context);
+  }
+
+  const data: any = await parseBody(output.body, context);
+  let contents: any = {};
+  contents = _json(data);
+  const response: OptionalInputOutputCommandOutput = {
+    $metadata: deserializeMetadata(output),
+    ...contents,
+  };
+  return response;
+};
+
+/**
+ * deserializeRpcv2cborRecursiveShapesCommand
+ */
+export const de_RecursiveShapesCommand = async (
+  output: __HttpResponse,
+  context: __SerdeContext
+): Promise<RecursiveShapesCommandOutput> => {
+  cr(output);
+  if (output.statusCode >= 300) {
+    return de_CommandError(output, context);
+  }
+
+  const data: any = await parseBody(output.body, context);
+  let contents: any = {};
+  contents = de_RecursiveShapesInputOutput(data, context);
+  const response: RecursiveShapesCommandOutput = {
+    $metadata: deserializeMetadata(output),
+    ...contents,
+  };
+  return response;
+};
+
+/**
+ * deserializeRpcv2cborRpcV2CborDenseMapsCommand
+ */
+export const de_RpcV2CborDenseMapsCommand = async (
+  output: __HttpResponse,
+  context: __SerdeContext
+): Promise<RpcV2CborDenseMapsCommandOutput> => {
+  cr(output);
+  if (output.statusCode >= 300) {
+    return de_CommandError(output, context);
+  }
+
+  const data: any = await parseBody(output.body, context);
+  let contents: any = {};
+  contents = _json(data);
+  const response: RpcV2CborDenseMapsCommandOutput = {
+    $metadata: deserializeMetadata(output),
+    ...contents,
+  };
+  return response;
+};
+
+/**
+ * deserializeRpcv2cborRpcV2CborListsCommand
+ */
+export const de_RpcV2CborListsCommand = async (
+  output: __HttpResponse,
+  context: __SerdeContext
+): Promise<RpcV2CborListsCommandOutput> => {
+  cr(output);
+  if (output.statusCode >= 300) {
+    return de_CommandError(output, context);
+  }
+
+  const data: any = await parseBody(output.body, context);
+  let contents: any = {};
+  contents = de_RpcV2CborListInputOutput(data, context);
+  const response: RpcV2CborListsCommandOutput = {
+    $metadata: deserializeMetadata(output),
+    ...contents,
+  };
+  return response;
+};
+
+/**
+ * deserializeRpcv2cborRpcV2CborSparseMapsCommand
+ */
+export const de_RpcV2CborSparseMapsCommand = async (
+  output: __HttpResponse,
+  context: __SerdeContext
+): Promise<RpcV2CborSparseMapsCommandOutput> => {
+  cr(output);
+  if (output.statusCode >= 300) {
+    return de_CommandError(output, context);
+  }
+
+  const data: any = await parseBody(output.body, context);
+  let contents: any = {};
+  contents = de_RpcV2CborSparseMapsInputOutput(data, context);
+  const response: RpcV2CborSparseMapsCommandOutput = {
+    $metadata: deserializeMetadata(output),
+    ...contents,
+  };
+  return response;
```

Comment:
All these de_Command functions look basically the same, other than the function used to assign to `contents`. Is there something else that could happen here or could refactor out the common code?

Reply:
I believe shortening this may be out of scope, because they may end up identical, but have the potential to be different based on certain model parameters. It's a general issue affecting codegen of all protocols.

Reply:
Good point. Something to investigate in the future I guess.